pkg/action/install_test.go

```diff
@@ -798,3 +798,46 @@ func TestInstallWithSystemLabels(t *testing.T) {
 
 	is.Equal(fmt.Errorf("user supplied labels contains system reserved label name. System labels: %+v", driver.GetSystemLabels()), err)
 }
+
+func TestInstallWithTakeOwnershipOption(t *testing.T) {
+	is := assert.New(t)
+	req := require.New(t)
+
+	instAction := installAction(t)
+	instAction.ClientOnly = false
+	instAction.IsUpgrade = false
+	instAction.TakeOwnership = true
+	failer := instAction.cfg.KubeClient.(*kubefake.FailingKubeClient)
+	failer.BuildDummy = true
+	instAction.cfg.KubeClient = failer
+
+	vals := map[string]interface{}{}
+	ctx, done := context.WithCancel(context.Background())
+	res, err := instAction.RunWithContext(ctx, buildChart(), vals)
+	if err != nil {
+		t.Fatalf("Failed install: %s", err)
+	}
+	is.Equal(res.Name, "test-install-release", "Expected release name.")
+	is.Equal(res.Namespace, "spaced")
+
+	rel, err := instAction.cfg.Releases.Get(res.Name, res.Version)
```

Comment:
Overall, IIUC the take-ownership flag is used to update existing k8s resources to be owned by helm using the labels/annotations. So the test should create some resources and test that when take-ownership is set helm is able to install and the resources now have the helm mgmt metadata.

Additionally, it might be good to have a negative test where we test without take-ownership and verify that the install does not work in that case

Reply:
This function shows what that metadata is: https://github.com/helm/helm/blob/main/pkg/action/validate.go#L94